src/fsharp/PatternMatchCompilation.fs

```diff
@@ -227,94 +237,109 @@ let RefuteDiscrimSet g m path discrims =
-   match r1,r2 with
-   | Expr.Val(vref,_,_), other | other, Expr.Val(vref,_,_) when vref.LogicalName = "_" -> other 
-   | Expr.Val(vref,_,_), other | other, Expr.Val(vref,_,_) when vref.LogicalName = notNullText -> other 
-   | Expr.Val(vref,_,_), other | other, Expr.Val(vref,_,_) when vref.LogicalName = otherSubtypeText -> other 
+    match r1,r2 with
+    | Expr.Val(vref,_,_), other | other, Expr.Val(vref,_,_) when vref.LogicalName = "_" -> other
+    | Expr.Val(vref,_,_), other | other, Expr.Val(vref,_,_) when vref.LogicalName = notNullText -> other 
+    | Expr.Val(vref,_,_), other | other, Expr.Val(vref,_,_) when vref.LogicalName = otherSubtypeText -> other 
 
-   | Expr.Op((TOp.ExnConstr(ecref1) as op1), tinst1,flds1,m1), Expr.Op(TOp.ExnConstr(ecref2), _,flds2,_) when tyconRefEq g ecref1 ecref2 -> 
+    | Expr.Op((TOp.ExnConstr(ecref1) as op1), tinst1,flds1,m1), Expr.Op(TOp.ExnConstr(ecref2), _,flds2,_) when tyconRefEq g ecref1 ecref2 -> 
         Expr.Op(op1, tinst1,List.map2 (CombineRefutations g) flds1 flds2,m1)
 
-   | Expr.Op((TOp.UnionCase(ucref1) as op1), tinst1,flds1,m1), 
-     Expr.Op(TOp.UnionCase(ucref2), _,flds2,_)  -> 
-       if g.unionCaseRefEq ucref1 ucref2 then 
-           Expr.Op(op1, tinst1,List.map2 (CombineRefutations g) flds1 flds2,m1)
-       (* Choose the greater of the two ucrefs based on name ordering *)
-       elif ucref1.CaseName < ucref2.CaseName then 
-           r2
-       else 
-           r1
+    | Expr.Op((TOp.UnionCase(ucref1) as op1), tinst1,flds1,m1), 
+      Expr.Op(TOp.UnionCase(ucref2), _,flds2,_)  -> 
+        if g.unionCaseRefEq ucref1 ucref2 then 
+            Expr.Op(op1, tinst1,List.map2 (CombineRefutations g) flds1 flds2,m1)
+         (* Choose the greater of the two ucrefs based on name ordering *)
+        elif ucref1.CaseName < ucref2.CaseName then 
+            r2
+        else 
+            r1
         
-   | Expr.Op(op1, tinst1,flds1,m1), Expr.Op(_, _,flds2,_) -> 
-        Expr.Op(op1, tinst1,List.map2 (CombineRefutations g) flds1 flds2,m1)
+    | Expr.Op(op1, tinst1,flds1,m1), Expr.Op(_, _,flds2,_) -> 
+         Expr.Op(op1, tinst1,List.map2 (CombineRefutations g) flds1 flds2,m1)
         
-   | Expr.Const(c1, m1, ty1), Expr.Const(c2,_,_) -> 
-       let c12 = 
-
-           // Make sure longer strings are greater, not the case in the default ordinal comparison
-           // This is needed because the individual counter examples make longer strings
-           let MaxStrings s1 s2 = 
-               let c = compare (String.length s1) (String.length s2)
-               if c < 0 then s2 
-               elif c > 0 then s1 
-               elif s1 < s2 then s2 
-               else s1
-               
-           match c1,c2 with 
-           | Const.String(s1), Const.String(s2) -> Const.String(MaxStrings s1 s2)
-           | Const.Decimal(s1), Const.Decimal(s2) -> Const.Decimal(max s1 s2)
-           | _ -> max c1 c2 
+    | Expr.Const(c1, m1, ty1), Expr.Const(c2,_,_) -> 
```

Comment:
It may be best to remove all these whitespace changes - otherwise this PR will get "blame" for every line. (I don't think it's particularly necessary to cleanup the whitespace here, and better to keep the PR minimal)